// fw-form-builder/dist/api/RatingCountsResponseApi.js
"use strict";

/**
 * FormBuilder
 * FormBuilder API
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
var ApiClient = require("../ApiClient");

var Error = require('../model/Error');
/**
* RatingCountsResponse service.
* @module api/RatingCountsResponseApi
* @version 0.0.0-10
*/


module.exports = class RatingCountsResponseApi {
  /**
  * Constructs a new RatingCountsResponseApi. 
  * @alias module:api/RatingCountsResponseApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  constructor(apiClient) {
    this.apiClient = apiClient || ApiClient.instance;
  }
  /**
   * Fetch ratings count for a form
   * Fetch ratings count for a form
   * @param {String} uuid Unique Identifier of a Form
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link Object} and HTTP response
   */


  getRatingCountsForFormWithHttpInfo(uuid) {
    let postBody = null; // verify the required parameter 'uuid' is set

    if (uuid === undefined || uuid === null) {
      throw new Error("Missing the required parameter 'uuid' when calling getRatingCountsForForm");
    }

    let pathParams = {
      'uuid': uuid
    };
    let queryParams = {};
    let headerParams = {};
    let formParams = {};
    let authNames = [];
    let contentTypes = [];
    let accepts = ['application/json'];
    let returnType = Object;
    return this.apiClient.callApi('/forms/{uuid}/ratings', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
  }
  /**
   * Fetch ratings count for a form
   * Fetch ratings count for a form
   * @param {String} uuid Unique Identifier of a Form
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link Object}
   */


  getRatingCountsForForm(uuid) {
    return this.getRatingCountsForFormWithHttpInfo(uuid).then(function (response_and_data) {
      return response_and_data;
    });
  }

};
