// fw-form-builder/dist/model/Error.js
"use strict";

/**
 * FormBuilder
 * FormBuilder API
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
var ApiClient = require('../ApiClient');
/**
 * The Error model module.
 * @module model/Error
 * @version 0.0.0-10
 */


class Error {
  /**
   * Constructs a new <code>Error</code>.
   * @alias module:model/Error
   */
  constructor() {
    Error.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  static initialize(obj) {}
  /**
   * Constructs a <code>Error</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Error} obj Optional instance to populate.
   * @return {module:model/Error} The populated <code>Error</code> instance.
   */


  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new Error();

      if (data.hasOwnProperty('message')) {
        obj['message'] = ApiClient.convertToType(data['message'], 'String');
      }
    }

    return obj;
  }
  /**
       * Returns Type of the error
       * @return {String}
       */


  getMessage() {
    return this.message;
  }
  /**
   * Sets Type of the error
   * @param {String} message Type of the error
   */


  setMessage(message) {
    this['message'] = message;
  }

}
/**
 * Type of the error
 * @member {String} message
 */


Error.prototype['message'] = undefined;
module.exports = Error;
