// fw-form-builder/dist/model/Form.js
"use strict";

/**
 * FormBuilder
 * FormBuilder API
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
var ApiClient = require('../ApiClient');
/**
 * The Form model module.
 * @module model/Form
 * @version 0.0.0-10
 */


class Form {
  /**
   * Constructs a new <code>Form</code>.
   * @alias module:model/Form
   */
  constructor() {
    Form.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  static initialize(obj) {}
  /**
   * Constructs a <code>Form</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Form} obj Optional instance to populate.
   * @return {module:model/Form} The populated <code>Form</code> instance.
   */


  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new Form();

      if (data.hasOwnProperty('id')) {
        obj['id'] = ApiClient.convertToType(data['id'], 'Number');
      }

      if (data.hasOwnProperty('lego_record_id')) {
        obj['lego_record_id'] = ApiClient.convertToType(data['lego_record_id'], 'String');
      }

      if (data.hasOwnProperty('name')) {
        obj['name'] = ApiClient.convertToType(data['name'], 'String');
      }

      if (data.hasOwnProperty('form_id')) {
        obj['form_id'] = ApiClient.convertToType(data['form_id'], 'String');
      }

      if (data.hasOwnProperty('lego_entity_id')) {
        obj['lego_entity_id'] = ApiClient.convertToType(data['lego_entity_id'], 'Number');
      }

      if (data.hasOwnProperty('form_json')) {
        obj['form_json'] = ApiClient.convertToType(data['form_json'], 'String');
      }

      if (data.hasOwnProperty('state')) {
        obj['state'] = ApiClient.convertToType(data['state'], 'String');
      }

      if (data.hasOwnProperty('flags')) {
        obj['flags'] = ApiClient.convertToType(data['flags'], 'Number');
      }

      if (data.hasOwnProperty('enable_notifications')) {
        obj['enable_notifications'] = ApiClient.convertToType(data['enable_notifications'], 'String');
      }

      if (data.hasOwnProperty('account_id')) {
        obj['account_id'] = ApiClient.convertToType(data['account_id'], 'Number');
      }

      if (data.hasOwnProperty('product')) {
        obj['product'] = ApiClient.convertToType(data['product'], 'String');
      }

      if (data.hasOwnProperty('schedules_created')) {
        obj['schedules_created'] = ApiClient.convertToType(data['schedules_created'], 'Number');
      }

      if (data.hasOwnProperty('rating_counts_record_id')) {
        obj['rating_counts_record_id'] = ApiClient.convertToType(data['rating_counts_record_id'], 'String');
      }

      if (data.hasOwnProperty('customer_details_entity_id')) {
        obj['customer_details_entity_id'] = ApiClient.convertToType(data['customer_details_entity_id'], 'Number');
      }

      if (data.hasOwnProperty('webhook_request_type')) {
        obj['webhook_request_type'] = ApiClient.convertToType(data['webhook_request_type'], 'String');
      }

      if (data.hasOwnProperty('webhook_url')) {
        obj['webhook_url'] = ApiClient.convertToType(data['webhook_url'], 'String');
      }

      if (data.hasOwnProperty('webhook_authorization')) {
        obj['webhook_authorization'] = ApiClient.convertToType(data['webhook_authorization'], 'String');
      }

      if (data.hasOwnProperty('webhook_body')) {
        obj['webhook_body'] = ApiClient.convertToType(data['webhook_body'], 'String');
      }
    }

    return obj;
  }
  /**
       * @return {Number}
       */


  getId() {
    return this.id;
  }
  /**
   * @param {Number} id
   */


  setId(id) {
    this['id'] = id;
  }
  /**
       * @return {String}
       */


  getLegoRecordId() {
    return this.lego_record_id;
  }
  /**
   * @param {String} legoRecordId
   */


  setLegoRecordId(legoRecordId) {
    this['lego_record_id'] = legoRecordId;
  }
  /**
       * @return {String}
       */


  getName() {
    return this.name;
  }
  /**
   * @param {String} name
   */


  setName(name) {
    this['name'] = name;
  }
  /**
       * Returns UUID for the form
       * @return {String}
       */


  getFormId() {
    return this.form_id;
  }
  /**
   * Sets UUID for the form
   * @param {String} formId UUID for the form
   */


  setFormId(formId) {
    this['form_id'] = formId;
  }
  /**
       * Returns Lego Entity ID
       * @return {Number}
       */


  getLegoEntityId() {
    return this.lego_entity_id;
  }
  /**
   * Sets Lego Entity ID
   * @param {Number} legoEntityId Lego Entity ID
   */


  setLegoEntityId(legoEntityId) {
    this['lego_entity_id'] = legoEntityId;
  }
  /**
       * Returns UI form JSON for form builder
       * @return {String}
       */


  getFormJson() {
    return this.form_json;
  }
  /**
   * Sets UI form JSON for form builder
   * @param {String} formJson UI form JSON for form builder
   */


  setFormJson(formJson) {
    this['form_json'] = formJson;
  }
  /**
       * @return {module:model/Form.StateEnum}
       */


  getState() {
    return this.state;
  }
  /**
   * @param {module:model/Form.StateEnum} state
   */


  setState(state) {
    this['state'] = state;
  }
  /**
       * @return {Number}
       */


  getFlags() {
    return this.flags;
  }
  /**
   * @param {Number} flags
   */


  setFlags(flags) {
    this['flags'] = flags;
  }
  /**
       * @return {String}
       */


  getEnableNotifications() {
    return this.enable_notifications;
  }
  /**
   * @param {String} enableNotifications
   */


  setEnableNotifications(enableNotifications) {
    this['enable_notifications'] = enableNotifications;
  }
  /**
       * @return {Number}
       */


  getAccountId() {
    return this.account_id;
  }
  /**
   * @param {Number} accountId
   */


  setAccountId(accountId) {
    this['account_id'] = accountId;
  }
  /**
       * @return {module:model/Form.ProductEnum}
       */


  getProduct() {
    return this.product;
  }
  /**
   * @param {module:model/Form.ProductEnum} product
   */


  setProduct(product) {
    this['product'] = product;
  }
  /**
       * @return {Number}
       */


  getSchedulesCreated() {
    return this.schedules_created;
  }
  /**
   * @param {Number} schedulesCreated
   */


  setSchedulesCreated(schedulesCreated) {
    this['schedules_created'] = schedulesCreated;
  }
  /**
       * @return {String}
       */


  getRatingCountsRecordId() {
    return this.rating_counts_record_id;
  }
  /**
   * @param {String} ratingCountsRecordId
   */


  setRatingCountsRecordId(ratingCountsRecordId) {
    this['rating_counts_record_id'] = ratingCountsRecordId;
  }
  /**
       * @return {Number}
       */


  getCustomerDetailsEntityId() {
    return this.customer_details_entity_id;
  }
  /**
   * @param {Number} customerDetailsEntityId
   */


  setCustomerDetailsEntityId(customerDetailsEntityId) {
    this['customer_details_entity_id'] = customerDetailsEntityId;
  }
  /**
       * @return {String}
       */


  getWebhookRequestType() {
    return this.webhook_request_type;
  }
  /**
   * @param {String} webhookRequestType
   */


  setWebhookRequestType(webhookRequestType) {
    this['webhook_request_type'] = webhookRequestType;
  }
  /**
       * @return {String}
       */


  getWebhookUrl() {
    return this.webhook_url;
  }
  /**
   * @param {String} webhookUrl
   */


  setWebhookUrl(webhookUrl) {
    this['webhook_url'] = webhookUrl;
  }
  /**
       * @return {String}
       */


  getWebhookAuthorization() {
    return this.webhook_authorization;
  }
  /**
   * @param {String} webhookAuthorization
   */


  setWebhookAuthorization(webhookAuthorization) {
    this['webhook_authorization'] = webhookAuthorization;
  }
  /**
       * @return {String}
       */


  getWebhookBody() {
    return this.webhook_body;
  }
  /**
   * @param {String} webhookBody
   */


  setWebhookBody(webhookBody) {
    this['webhook_body'] = webhookBody;
  }

}
/**
 * @member {Number} id
 */


Form.prototype['id'] = undefined;
/**
 * @member {String} lego_record_id
 */

Form.prototype['lego_record_id'] = undefined;
/**
 * @member {String} name
 */

Form.prototype['name'] = undefined;
/**
 * UUID for the form
 * @member {String} form_id
 */

Form.prototype['form_id'] = undefined;
/**
 * Lego Entity ID
 * @member {Number} lego_entity_id
 */

Form.prototype['lego_entity_id'] = undefined;
/**
 * UI form JSON for form builder
 * @member {String} form_json
 */

Form.prototype['form_json'] = undefined;
/**
 * @member {module:model/Form.StateEnum} state
 */

Form.prototype['state'] = undefined;
/**
 * @member {Number} flags
 */

Form.prototype['flags'] = undefined;
/**
 * @member {String} enable_notifications
 */

Form.prototype['enable_notifications'] = undefined;
/**
 * @member {Number} account_id
 */

Form.prototype['account_id'] = undefined;
/**
 * @member {module:model/Form.ProductEnum} product
 */

Form.prototype['product'] = undefined;
/**
 * @member {Number} schedules_created
 */

Form.prototype['schedules_created'] = undefined;
/**
 * @member {String} rating_counts_record_id
 */

Form.prototype['rating_counts_record_id'] = undefined;
/**
 * @member {Number} customer_details_entity_id
 */

Form.prototype['customer_details_entity_id'] = undefined;
/**
 * @member {String} webhook_request_type
 */

Form.prototype['webhook_request_type'] = undefined;
/**
 * @member {String} webhook_url
 */

Form.prototype['webhook_url'] = undefined;
/**
 * @member {String} webhook_authorization
 */

Form.prototype['webhook_authorization'] = undefined;
/**
 * @member {String} webhook_body
 */

Form.prototype['webhook_body'] = undefined;
/**
 * Allowed values for the <code>state</code> property.
 * @enum {String}
 * @readonly
 */

Form['StateEnum'] = {
  /**
   * value: "live"
   * @const
   */
  "live": "live",

  /**
   * value: "disabled"
   * @const
   */
  "disabled": "disabled"
};
/**
 * Allowed values for the <code>product</code> property.
 * @enum {String}
 * @readonly
 */

Form['ProductEnum'] = {
  /**
   * value: "FRESHDESK"
   * @const
   */
  "FRESHDESK": "FRESHDESK",

  /**
   * value: "FRESHSALES"
   * @const
   */
  "FRESHSALES": "FRESHSALES",

  /**
   * value: "FRESHSERVICE"
   * @const
   */
  "FRESHSERVICE": "FRESHSERVICE",

  /**
   * value: "FRESHTEAM"
   * @const
   */
  "FRESHTEAM": "FRESHTEAM"
};
module.exports = Form;
