// fw-form-builder/dist/model/SurveySchedule.js
"use strict";

/**
 * FormBuilder
 * FormBuilder API
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
var ApiClient = require('../ApiClient');
/**
 * The SurveySchedule model module.
 * @module model/SurveySchedule
 * @version 0.0.0-10
 */


class SurveySchedule {
  /**
   * Constructs a new <code>SurveySchedule</code>.
   * @alias module:model/SurveySchedule
   */
  constructor() {
    SurveySchedule.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  static initialize(obj) {}
  /**
   * Constructs a <code>SurveySchedule</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/SurveySchedule} obj Optional instance to populate.
   * @return {module:model/SurveySchedule} The populated <code>SurveySchedule</code> instance.
   */


  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new SurveySchedule();

      if (data.hasOwnProperty('id')) {
        obj['id'] = ApiClient.convertToType(data['id'], 'Number');
      }

      if (data.hasOwnProperty('first_survey_date')) {
        obj['first_survey_date'] = ApiClient.convertToType(data['first_survey_date'], 'String');
      }

      if (data.hasOwnProperty('frequency_in_days')) {
        obj['frequency_in_days'] = ApiClient.convertToType(data['frequency_in_days'], 'Number');
      }

      if (data.hasOwnProperty('schdeuled_time')) {
        obj['schdeuled_time'] = ApiClient.convertToType(data['schdeuled_time'], 'String');
      }

      if (data.hasOwnProperty('customer_emails')) {
        obj['customer_emails'] = ApiClient.convertToType(data['customer_emails'], 'String');
      }
    }

    return obj;
  }
  /**
       * @return {Number}
       */


  getId() {
    return this.id;
  }
  /**
   * @param {Number} id
   */


  setId(id) {
    this['id'] = id;
  }
  /**
       * @return {String}
       */


  getFirstSurveyDate() {
    return this.first_survey_date;
  }
  /**
   * @param {String} firstSurveyDate
   */


  setFirstSurveyDate(firstSurveyDate) {
    this['first_survey_date'] = firstSurveyDate;
  }
  /**
       * @return {Number}
       */


  getFrequencyInDays() {
    return this.frequency_in_days;
  }
  /**
   * @param {Number} frequencyInDays
   */


  setFrequencyInDays(frequencyInDays) {
    this['frequency_in_days'] = frequencyInDays;
  }
  /**
       * @return {String}
       */


  getSchdeuledTime() {
    return this.schdeuled_time;
  }
  /**
   * @param {String} schdeuledTime
   */


  setSchdeuledTime(schdeuledTime) {
    this['schdeuled_time'] = schdeuledTime;
  }
  /**
       * @return {String}
       */


  getCustomerEmails() {
    return this.customer_emails;
  }
  /**
   * @param {String} customerEmails
   */


  setCustomerEmails(customerEmails) {
    this['customer_emails'] = customerEmails;
  }

}
/**
 * @member {Number} id
 */


SurveySchedule.prototype['id'] = undefined;
/**
 * @member {String} first_survey_date
 */

SurveySchedule.prototype['first_survey_date'] = undefined;
/**
 * @member {Number} frequency_in_days
 */

SurveySchedule.prototype['frequency_in_days'] = undefined;
/**
 * @member {String} schdeuled_time
 */

SurveySchedule.prototype['schdeuled_time'] = undefined;
/**
 * @member {String} customer_emails
 */

SurveySchedule.prototype['customer_emails'] = undefined;
module.exports = SurveySchedule;
