// fw-form-builder/dist/model/SurveyScheduleResponse.js
"use strict";

/**
 * FormBuilder
 * FormBuilder API
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
var ApiClient = require('../ApiClient');
/**
 * The SurveyScheduleResponse model module.
 * @module model/SurveyScheduleResponse
 * @version 0.0.0-10
 */


class SurveyScheduleResponse {
  /**
   * Constructs a new <code>SurveyScheduleResponse</code>.
   * @alias module:model/SurveyScheduleResponse
   */
  constructor() {
    SurveyScheduleResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  static initialize(obj) {}
  /**
   * Constructs a <code>SurveyScheduleResponse</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/SurveyScheduleResponse} obj Optional instance to populate.
   * @return {module:model/SurveyScheduleResponse} The populated <code>SurveyScheduleResponse</code> instance.
   */


  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new SurveyScheduleResponse();

      if (data.hasOwnProperty('id')) {
        obj['id'] = ApiClient.convertToType(data['id'], 'String');
      }

      if (data.hasOwnProperty('form_id')) {
        obj['form_id'] = ApiClient.convertToType(data['form_id'], 'String');
      }

      if (data.hasOwnProperty('schedules_created')) {
        obj['schedules_created'] = ApiClient.convertToType(data['schedules_created'], 'Number');
      }
    }

    return obj;
  }
  /**
       * @return {String}
       */


  getId() {
    return this.id;
  }
  /**
   * @param {String} id
   */


  setId(id) {
    this['id'] = id;
  }
  /**
       * @return {String}
       */


  getFormId() {
    return this.form_id;
  }
  /**
   * @param {String} formId
   */


  setFormId(formId) {
    this['form_id'] = formId;
  }
  /**
       * @return {Number}
       */


  getSchedulesCreated() {
    return this.schedules_created;
  }
  /**
   * @param {Number} schedulesCreated
   */


  setSchedulesCreated(schedulesCreated) {
    this['schedules_created'] = schedulesCreated;
  }

}
/**
 * @member {String} id
 */


SurveyScheduleResponse.prototype['id'] = undefined;
/**
 * @member {String} form_id
 */

SurveyScheduleResponse.prototype['form_id'] = undefined;
/**
 * @member {Number} schedules_created
 */

SurveyScheduleResponse.prototype['schedules_created'] = undefined;
module.exports = SurveyScheduleResponse;
