// fw-form-builder/dist/model/SurveyScheduleRequest.js
"use strict";

/**
 * FormBuilder
 * FormBuilder API
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
var ApiClient = require('../ApiClient');
/**
 * The SurveyScheduleRequest model module.
 * @module model/SurveyScheduleRequest
 * @version 0.0.0-10
 */


class SurveyScheduleRequest {
  /**
   * Constructs a new <code>SurveyScheduleRequest</code>.
   * @alias module:model/SurveyScheduleRequest
   */
  constructor() {
    SurveyScheduleRequest.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  static initialize(obj) {}
  /**
   * Constructs a <code>SurveyScheduleRequest</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/SurveyScheduleRequest} obj Optional instance to populate.
   * @return {module:model/SurveyScheduleRequest} The populated <code>SurveyScheduleRequest</code> instance.
   */


  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new SurveyScheduleRequest();

      if (data.hasOwnProperty('id')) {
        obj['id'] = ApiClient.convertToType(data['id'], 'String');
      }

      if (data.hasOwnProperty('form_id')) {
        obj['form_id'] = ApiClient.convertToType(data['form_id'], 'String');
      }

      if (data.hasOwnProperty('first_survey_date')) {
        obj['first_survey_date'] = ApiClient.convertToType(data['first_survey_date'], 'String');
      }

      if (data.hasOwnProperty('frequency_in_days')) {
        obj['frequency_in_days'] = ApiClient.convertToType(data['frequency_in_days'], 'Number');
      }

      if (data.hasOwnProperty('schdeuled_time')) {
        obj['schdeuled_time'] = ApiClient.convertToType(data['schdeuled_time'], 'String');
      }

      if (data.hasOwnProperty('customer_emails')) {
        obj['customer_emails'] = ApiClient.convertToType(data['customer_emails'], 'String');
      }

      if (data.hasOwnProperty('survey_url')) {
        obj['survey_url'] = ApiClient.convertToType(data['survey_url'], 'String');
      }

      if (data.hasOwnProperty('existing_schedules')) {
        obj['existing_schedules'] = ApiClient.convertToType(data['existing_schedules'], 'Number');
      }
    }

    return obj;
  }
  /**
       * @return {String}
       */


  getId() {
    return this.id;
  }
  /**
   * @param {String} id
   */


  setId(id) {
    this['id'] = id;
  }
  /**
       * @return {String}
       */


  getFormId() {
    return this.form_id;
  }
  /**
   * @param {String} formId
   */


  setFormId(formId) {
    this['form_id'] = formId;
  }
  /**
       * @return {String}
       */


  getFirstSurveyDate() {
    return this.first_survey_date;
  }
  /**
   * @param {String} firstSurveyDate
   */


  setFirstSurveyDate(firstSurveyDate) {
    this['first_survey_date'] = firstSurveyDate;
  }
  /**
       * @return {Number}
       */


  getFrequencyInDays() {
    return this.frequency_in_days;
  }
  /**
   * @param {Number} frequencyInDays
   */


  setFrequencyInDays(frequencyInDays) {
    this['frequency_in_days'] = frequencyInDays;
  }
  /**
       * @return {String}
       */


  getSchdeuledTime() {
    return this.schdeuled_time;
  }
  /**
   * @param {String} schdeuledTime
   */


  setSchdeuledTime(schdeuledTime) {
    this['schdeuled_time'] = schdeuledTime;
  }
  /**
       * @return {String}
       */


  getCustomerEmails() {
    return this.customer_emails;
  }
  /**
   * @param {String} customerEmails
   */


  setCustomerEmails(customerEmails) {
    this['customer_emails'] = customerEmails;
  }
  /**
       * @return {String}
       */


  getSurveyUrl() {
    return this.survey_url;
  }
  /**
   * @param {String} surveyUrl
   */


  setSurveyUrl(surveyUrl) {
    this['survey_url'] = surveyUrl;
  }
  /**
       * @return {Number}
       */


  getExistingSchedules() {
    return this.existing_schedules;
  }
  /**
   * @param {Number} existingSchedules
   */


  setExistingSchedules(existingSchedules) {
    this['existing_schedules'] = existingSchedules;
  }

}
/**
 * @member {String} id
 */


SurveyScheduleRequest.prototype['id'] = undefined;
/**
 * @member {String} form_id
 */

SurveyScheduleRequest.prototype['form_id'] = undefined;
/**
 * @member {String} first_survey_date
 */

SurveyScheduleRequest.prototype['first_survey_date'] = undefined;
/**
 * @member {Number} frequency_in_days
 */

SurveyScheduleRequest.prototype['frequency_in_days'] = undefined;
/**
 * @member {String} schdeuled_time
 */

SurveyScheduleRequest.prototype['schdeuled_time'] = undefined;
/**
 * @member {String} customer_emails
 */

SurveyScheduleRequest.prototype['customer_emails'] = undefined;
/**
 * @member {String} survey_url
 */

SurveyScheduleRequest.prototype['survey_url'] = undefined;
/**
 * @member {Number} existing_schedules
 */

SurveyScheduleRequest.prototype['existing_schedules'] = undefined;
module.exports = SurveyScheduleRequest;
